core/trino-main/src/test/java/io/trino/util/TestDateTimeUtils.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.util;
+
+import org.testng.annotations.Test;
+
+import java.time.DateTimeException;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.testng.Assert.assertEquals;
+
+public class TestDateTimeUtils
+{
+    @Test
+    public void testTryParseDateYyyyMmDd()
+    {
+        // valid dates
+        int days = DateTimeUtils.tryParseDateYyyyMmDd("1970-01-01");
+        assertEquals(0, days);
+        days = DateTimeUtils.tryParseDateYyyyMmDd("1970-02-01");
+        assertEquals(31, days);
+        days = DateTimeUtils.tryParseDateYyyyMmDd("1969-12-01");
+        assertEquals(-31, days);
```

Comment:
should tryParseDateYyyyMmDd support a negative year? e.g. `-1970-01-01`
also, add more test cases: e.g. min, max date that can be parsed, year zero, 29 feb leap year etc

Reply:
tryParseDateYyyyMmDd will skip format like -1970-01-01, and will fall back to the original implementation.
Anyway, I will add more test cases as you suggested

Reply:
more test cases added in cb70c37e453480e7b5891e117f63796e1ac450fb